### PR TITLE
Add GCP resource detection module

### PR DESCRIPTION
**Pull request by psx95**
**Add Google Cloud Resource Detection Module**

Moves the existing google cloud resource detectors from [GoogleCloudPlatform/opentelemetry-operations-java](https://github.com/GoogleCloudPlatform/opentelemetry-operations-java/tree/main/detectors/resources#resource-detectors-for-opentelemetry) to open-telemetry/opentelemetry-java-contrib.

**Existing Issue(s):**

 - Supports #1074

**Testing:**

 - `./gradlew assemble &&  ./gradlew :gcp-resources:test`
 -  Added unit tests to test the resource mapping.

**Documentation:**

Javadoc and README  explaining the purpose of the module added.

**Outstanding items:**

 - ~~Update the [component owners file](https://github.com/open-telemetry/opentelemetry-java-contrib/blob/fa5d16193b36d6732324cc8804fbe6708423dc7d/.github/component_owners.yml).~~

### CODE REVIEW

**Review by jsuereth — APPROVED** [review]

File: gcp-resources/README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+| k8s.namespace.name | | downward API | | | |
+| k8s.container.name | | hardcoded | | | |
+| k8s.cluster.name | | auto | | | |
+| faas.name | | | environment var | environment var | environment var |
+| faas.version | | | environment var | environment var | environment var |
+| faas.instance | | | auto | auto | environment var |
+
+## Downward API
+
+For GKE applications, some values most be passed via the environment variable using k8s
```
Comment (SylvainJuge) [inline gcp-resources/README.md:27]:
```suggestion
For GKE applications, some values must be passed via the environment variable using k8s
```

File: gcp-resources/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+description = "OpenTelemetry GCP Resources Support"
+otelJava.moduleName.set("io.opentelemetry.contrib.gcp.resource")
+
+dependencies {
+  api("io.opentelemetry:opentelemetry-api")
+  api("io.opentelemetry:opentelemetry-sdk")
+
+  // Provides GCP resource detection support
+  implementation("com.google.cloud.opentelemetry:detector-resources-support:0.27.0")
```
Comment (SylvainJuge) [inline gcp-resources/build.gradle.kts:15]: [question] would this be automatically updated through dependabot (or similar) when there is a new release ?

File: gcp-resources/src/test/java/io/opentelemetry/contrib/gcp/resource/GCPResourceProviderTest.java [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+  @Test
+  public void testGceResourceAttributesMapping() {
+    GCPPlatformDetector mockDetector = Mockito.mock(GCPPlatformDetector.class);
+    DetectedPlatform mockPlatform = generateMockGcePlatform();
+    Mockito.when(mockDetector.detectPlatform()).thenReturn(mockPlatform);
+
+    Resource gotResource = new GCPResourceProvider(mockDetector).createResource(mockConfigProps);
+    Mockito.verify(mockPlatform, Mockito.times(1)).getProjectId();
+
+    assertEquals(
```
Comment (SylvainJuge) [inline gcp-resources/src/test/java/io/opentelemetry/contrib/gcp/resource/GCPResourceProviderTest.java:158]: [minor nitpick] maybe switch to using assertJ `assertThat` here as it's widely used elsewhere in the project, also it provides a convenient syntax for checking map entries.

File: gcp-resources/src/test/java/io/opentelemetry/contrib/gcp/resource/GCPResourceProviderTest.java [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+  }
+
+  @Test
+  public void testGceResourceAttributesMapping() {
+    GCPPlatformDetector mockDetector = Mockito.mock(GCPPlatformDetector.class);
+    DetectedPlatform mockPlatform = generateMockGcePlatform();
+    Mockito.when(mockDetector.detectPlatform()).thenReturn(mockPlatform);
+
+    Resource gotResource = new GCPResourceProvider(mockDetector).createResource(mockConfigProps);
+    Mockito.verify(mockPlatform, Mockito.times(1)).getProjectId();
```
Comment (SylvainJuge) [inline gcp-resources/src/test/java/io/opentelemetry/contrib/gcp/resource/GCPResourceProviderTest.java:156]: [minor nitpick] maybe use static import for `Mockito.verify`.

**Review by SylvainJuge — APPROVED** [review]
Thanks a lot @psx95, that's awesome !

File: gcp-resources/build.gradle.kts (hunk 2 above)
Comment (trask) [reply gcp-resources/build.gradle.kts:15]: renovate will auto-update this by default

**Review by trask — APPROVED** [review]

**Comment by trask** [comment]
can you add the new module to here also?

https://github.com/open-telemetry/opentelemetry-java-contrib/blob/5000adfc2319cef17aeec50c052a593ebcf20778/.github/scripts/draft-change-log-entries.sh#L31-L32

**Comment by psx95** [comment]
> can you add the new module to here also?
> 
> https://github.com/open-telemetry/opentelemetry-java-contrib/blob/5000adfc2319cef17aeec50c052a593ebcf20778/.github/scripts/draft-change-log-entries.sh#L31-L32

Done! Thank you for the review !

**Pull request merged**